### PR TITLE
[BEAM-2708] Adds support for reading concatenated bzip2 files

### PR DESCRIPTION
**Pull request by chamikaramj**
Adds tests for concatenated gzip and bzip2 files.

Removes test 'test_model_textio_gzip_concatenated' in 'snippets_test.py' since it's actually hitting 'DummyReadTransform' and not testing this feature.

### CODE REVIEW

**Comment by chamikaramj** [comment]
R: @aaltay 

cc: @robertwb

File: sdks/python/apache_beam/io/textio_test.py [hunk 1]
```diff
@@ -415,6 +473,64 @@ def test_read_gzip(self):
...
+        file_name,
+        0, CompressionTypes.GZIP,
+        True, coders.StrUtf8Coder())
+    assert_that(pcoll, equal_to(lines))
+
+    with self.assertRaises(Exception):
+      pipeline.run()
+
+  def test_read_gzip_concat(self):
+    # gzip_file_name1 = temp_path_1 + '.gz'
```
Comment (aaltay) [inline sdks/python/apache_beam/io/textio_test.py]: Remove commented code.

File: sdks/python/apache_beam/io/filesystem.py [hunk 2]
```diff
@@ -187,29 +187,26 @@ def _fetch_to_internal_buffer(self, num_bytes):
...
-            self._decompressor = zlib.decompressobj(self._gzip_mask)
+            self._decompressor = (
+                bz2.BZ2Decompressor()
+                if self._compression_type == CompressionTypes.BZIP2
+                else zlib.decompressobj(self._gzip_mask))
             decompressed = self._decompressor.decompress(buf)
             self._read_buffer.write(decompressed)
             continue
         else:
+          # Gzip and bzip2 formats do not require flusing remaining data in the
```
Comment (aaltay) [inline sdks/python/apache_beam/io/filesystem.py]: `flusing` -> `flushing`

File: sdks/python/apache_beam/io/textio_test.py (hunk 1 above)
Comment (chamikaramj) [inline sdks/python/apache_beam/io/textio_test.py]: Done.

File: sdks/python/apache_beam/io/filesystem.py (hunk 2 above)
Comment (chamikaramj) [inline sdks/python/apache_beam/io/filesystem.py]: Done.

**Comment by chamikaramj** [comment]
Thanks.

**Pull request closed without merging**